### PR TITLE
I46 conductor weiym conductor

### PR DESCRIPTION
**Pull request by weiym97**
This is a pull request from i46-conductor-weiym-conductor to i46-conductor. There are five main changes in this branch.
(1) add logging to exe/qctool.cpp;
(2) Finish the minimal pipeline in the src/Conductor.cpp.
(3) add a bunch of whole pipeline test tools (written in python, using executable), which include pipelinetest_main.py, schema/PipelineCase.json, tests/pipelinetests.py
(4) add an overloading loadfile method to MdfDatasetManager so that the metadata could be changed in the whole pipeline test (this actually took a lot of time to debug).
(5) add more Presentation Contexts in the src/communication/Sender.cpp so that the postprocessed files can be sent successfully from the pipeline to the remote receiver.

Sorry there might be too many changes in this pull request. But all this branch has done is to ensure that a minimal pipeline for GE is working and minimally tested. This is exactly the aim of Issue 46.

Also, Ola has fixed the sanitizer issue in i46-conductor branch. So after merging i46-conductor-weiym-conductor to i46-conductor, we might be able to merge i46-conductor to the main branch, and close the Issue 46. 

*Several things need to be done next*:
(1) add more whole pipeline test cases, and run these whole pipeline tests automatically with python, which involves further debugging potentially. (Alexi is working on)
(2) further revise the "addPresentationContexts" in the src/sender.cpp so that all the files can be passed successfully (now 8/9 files succeeded in the whole pipeline)
(3) revise the logging, and create an individual logging file for each image.
(4) add the ImageEditor to the pipeline.
(5) Start to think about image localisation and segmentation tools.

### CODE REVIEW

File: pipeline_test.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+if __name__ == '__main__':
+    cmd_1 = "./build/exe/qctool --config-file='./schema/useCase.json'"
+    cmd_2 = "storescp 11113 --output-directory ./result/"
+    cmd_3 = "storescp 11114 --output-directory ./result_quarantine/"
+    cmd_4 = "storescu localhost 11112 DICOM_Images --scan-directories"
+    cmd = cmd_1 + " & " + cmd_2 + " & " + cmd_3 + " & " + cmd_4
+    os.system(cmd)
+
+    '''
+    Note: when we rerun the whole pipeline test,
```
Comment (martinjrobins) [inline pipeline_test.py:16]: can't you fill the processes at the end of the test?

**Review by martinjrobins — APPROVED** [review]
looks great, thanks @weiym97, I had a few comments mainly around overlapping python files, see below

**Review by ian-mcfarlane — APPROVED** [review]
This all looks great to me! Agreed with @martinjrobins that it's not clear to me why there are several pipelinetest.py files - do these have different uses?

File: exe/qctool.cpp [hunk 2]
```diff
@@ -61,20 +62,21 @@ int main(int argc, char **argv) {
 
     po::notify(var_map);
 
-    
-
+    set_root_logging("./result/pipeline_root_test.log", true);
```
[the remark this answers is not in the document]
Comment (weiym97) [reply exe/qctool.cpp:65]: Yeah, I've realized that! We'll open up a new issue on the logging, and at that branch, we will enable the user to change the log filenames in the command line argument.

File: pipeline_test.py (hunk 1 above)
Comment (weiym97) [reply pipeline_test.py:16]: I will figure out a way to shut down the pipeline, DCMTK sender & receiver using python in the future.

File: pipelinetest_main.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import os
```
[the remark this answers is not in the document]
Comment (weiym97) [reply pipelinetest_main.py:1]: The pipeline_test.py in the root directory is obsolete, I will delete it.

File: tests/pipelinetests.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import os
```
[the remark this answers is not in the document]
Comment (weiym97) [reply tests/pipelinetests.py:1]: This file contains the class and methods used by the pipelinetest_main.py. So I will clear the repeated codes in the pipelinetest_main.py, and use the class here.

**Comment by weiym97** [comment]
> This all looks great to me! Agreed with @martinjrobins that it's not clear to me why there are several pipelinetest.py files - do these have different uses?

Yeah, the pipeline_test.py in the root has expired. We will work on pipelinetest_main.py, and in that file, we will call the class and method in test/pipelinetests.py.

**Pull request merged**